### PR TITLE
fix bug in taxon card content tab when switch to other specie - 177659751

### PR DESCRIPTION
**Pull request by Hvar83**
The problem is in the taxon-card and it happens when i move between tabs and later i switch to another specie.

For example:
- go to taxon MX.44058
- move between tabs (at least open "occurrences" and "specimens")
- switch to other species by using dropdown menu in breadcrumbs. 
- if the next specie i open has less tabs than the previous, for example MX.44052, the content is not correct. 

2 components are opened at same time. At least i could see "specimens" and "occurrences" or "specimens" and "taxonomy" opened at same time.

### CODE REVIEW

File: projects/laji/src/app/+taxonomy/taxon/info-card/info-card.component.ts [hunk 1]
```diff
@@ -148,8 +148,12 @@ export class InfoCardComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   updateRoute(id: string, tab = this.activeTab, context = this.context, replaceUrl = false) {
-    this.selectedTab = tab;
+    if (this.activeTab !== 'endangerment' && this.activeTab !== 'invasive') {
+      this.loadedTabs.reset();
+    }
```
Comment (vixriihi) [inline projects/laji/src/app/+taxonomy/taxon/info-card/info-card.component.ts:153]: changing tab causes updateRoute so resetting loadedTabs on almost all tabs is not ideal. (idea with loaded tabs is to keep data for one specific taxon always in memory and not to re-fetch the same data on every tab change)

**Review by vixriihi — CHANGES_REQUESTED** [review]

**Review by vixriihi — APPROVED** [review]
LGTM

**Review by vixriihi — CHANGES_REQUESTED** [review]
not ready yet

**Comment by esko-piirainen** [comment]
There is a bug with this pull request found in beta - details in pivotal

**Comment by Blodir** [comment]
I have now fixed the bug that was mentioned in pivotal:

> https://177659751.dev.laji.fi/taxon/MX.66113/endangerment
> when you switch between taxa (using species dropdown in breadcrumb) and then back to the species the contents do not load

This was a separate, but somewhat related bug. Took me a very long time to figure it out despite the fix being only 2 lines long. I'll describe the issue here as I understand it.

Tabs component uses content projection to populate the content of each tab. I was falsely under the assumption that the change detector hierarchy would be linear ie.
```
<parent>
  <tabs>
    <tab>
      <content></content>
    </tab>
  </tabs>
</parent>
```
This is what the markup suggests. However content projection actually breaks the hierarchy and in fact `tabs`, `tab` and `content` are all direct children of `parent`. In order to run changedetection for `content` we actually have to call `detectChanges()` in `parent`.

After this fix everything seems to work fine.

**Comment by Blodir** [comment]
https://177659751.dev.laji.fi/

**Review by esko-piirainen — APPROVED** [review]
ok

**Pull request merged**